Std/Data/Fin/Lemmas.lean

```diff
@@ -29,7 +29,7 @@ theorem pos_iff_nonempty {n : Nat} : 0 < n ↔ Nonempty (Fin n) :=
 
 /-! ### coercions and constructions -/
 
-@[simp] protected theorem eta (a : Fin n) (h : a < n) : (⟨a, h⟩ : Fin n) = a := by cases a; rfl
+@[simp] protected theorem eta (a : Fin n) (h : a < n) : (⟨a, h⟩ : Fin n) = a := rfl
```

Comment:
It would be great to have a linter for this (`@[simp]` lemmas that could be proved by `rfl` but aren't). It is a headache for users when `simp` refuses to go into `dsimp` mode with lemmas like this.

Reply:
I've punted this to a [request](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/linter.20for.20.60rfl.60.20lemmas.3F/near/390817700) on zulip, it's too far a distraction for today!